Find images

jennyfolkesson opened the pull request:
Added support to read either a directory containing images or a number of subdirectories containing one image each.

I also added tests to the code, which I recommend doing moving forward for those who are so inclined. It really helps if the repo is intended to have many collaborators.

jennyfolkesson committed:
added ci yaml

jennyfolkesson committed:
added ci yaml

jennyfolkesson committed:
added ci yaml

jennyfolkesson committed:
added ci yaml

jennyfolkesson committed:
updated tests

jennyfolkesson committed:
updated tests

jennyfolkesson committed:
updated tests

jennyfolkesson committed:
updated tests

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

mattersoflight committed:
rename cli

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
debugging ci

jennyfolkesson committed:
changing to existing opencv version

jennyfolkesson committed:
changing to existing opencv version

jennyfolkesson committed:
changing to existing opencv version

jennyfolkesson committed:
changing to existing opencv version

jennyfolkesson committed:
changing to existing opencv version

jennyfolkesson committed:
changing to existing opencv version

jennyfolkesson committed:
changed test name to comply with pytest

jennyfolkesson committed:
changed test name to comply with pytest

jennyfolkesson committed:
changed test name to comply with pytest

jennyfolkesson committed:
changed test name to comply with pytest

jennyfolkesson committed:
changed test name to comply with pytest

jennyfolkesson committed:
changed test name to comply with pytest

jennyfolkesson committed:
changed test name to comply with pytest

jennyfolkesson committed:
changed test name to comply with pytest

jennyfolkesson committed:
adding testfixtures to ci

jennyfolkesson committed:
adding testfixtures to ci

jennyfolkesson committed:
adding testfixtures to ci

jennyfolkesson committed:
adding badge

jennyfolkesson committed:
adding badge

jennyfolkesson committed:
Merge branch 'ci_tests' of github.com:czbiohub/serology-COVID19 into ci_tests

jennyfolkesson committed:
added support for reading different types of well images

jennyfolkesson committed:
added support for reading different types of well images

jennyfolkesson committed:
added support for reading different types of well images

jennyfolkesson committed:
adding tests

jennyfolkesson committed:
adding tests

jennyfolkesson committed:
adding tests

jennyfolkesson committed:
comment failing test for now

jennyfolkesson committed:
comment failing test for now

jennyfolkesson committed:
comment failing test for now

mattersoflight committed:
add mutually exclusive arguments for pysero: extract_od, analyze_od

jennyfolkesson committed:
added test for micromanager style data

jennyfolkesson committed:
added test for micromanager style data

jennyfolkesson committed:
added test for micromanager style data

jennyfolkesson committed:
added scipy to requirements

jennyfolkesson committed:
added scipy to requirements

mattersoflight committed:
address the PR comments

mattersoflight committed:
Merge pull request #51 from czbiohub/rename_cli

give cli a bit more intuitive name.

mattersoflight committed:
Set theme jekyll-theme-slate

mattersoflight commented:
@jennyfolkesson I tried to run the pipeline with data acquired on Leica. I get following error in background estimation. 

<img width="1411" alt="image" src="https://user-images.githubusercontent.com/2934183/80540250-d19e5580-895d-11ea-961e-1eb999c958d8.png">

mattersoflight commented:
I get an error with the other micro-manager dataset too (vanilla ELISA)
<img width="1413" alt="image" src="https://user-images.githubusercontent.com/2934183/80540512-57ba9c00-895e-11ea-88dd-a709a17e21a2.png">

jennyfolkesson committed:
rebased and updated tests

jennyfolkesson committed:
removed unused requirements

jennyfolkesson commented:
@mattersoflight this branch has not been tested on Leica images, and probably can't until metadata is refactored.
The first error is probably due to Leica images are 512 x 512, and then they are cropped to who knows what and after that background estimation is done with a block size of 128, which is probably larger than the image at that point.
So we probably need to add background block size as one of the parameters that need to be adjusted to physical size instead of pixel size.

The second error is because I only looked at interpolation and registration workflows...sorry about that. I'll update well workflow right away!

jennyfolkesson committed:
updated well wf with io utils

jennyfolkesson committed:
updated well wf with io utils

jennyfolkesson committed:
updated well wf with io utils

jennyfolkesson committed:
adding back openpyxl

jennyfolkesson committed:
removed background instance generation from img processing

jennyfolkesson committed:
removed background instance generation from img processing

jennyfolkesson committed:
removed background instance generation from img processing

jennyfolkesson committed:
removed background instance generation from img processing

jennyfolkesson committed:
updated background estimation

jennyfolkesson committed:
updated background estimation

jennyfolkesson committed:
bg merge

mattersoflight committed:
Merge pull request #53 from czbiohub/ci_tests

Ci tests

mattersoflight committed:
update docs

mattersoflight committed:
Merge pull request #56 from czbiohub/documentation

update docs

mattersoflight committed:
Rename CONTRIBUTING.md to contributing.md

mattersoflight committed:
Update README.md

mattersoflight committed:
Update README.md

mattersoflight commented:
@jennyfolkesson I see that the error in well workflow came because of refactoring of the image io methods.

It is great that you started to refactor and are now adding tests in pytest format. 

Please implement the  following to make sure changes from this branch are useful in master:
* rebase onto current master and make sure that all tests use pytest library.
*  test array workflows in this branch with Scienion format `/Volumes/GoogleDrive/My Drive/ELISAarrayReader/images_scienion/2020-04-08-14-48-49-COVID_concentrationtest_April8_images` 
* test well workflows in this branch with `/Volumes/GoogleDrive/My Drive/ELISAarrayReader/images_leica/20200408_VanillaELISA_Leica2.5x_2` 
* If you like, curate the 5 'best images' and the 5 'worst images' for extract_od  workflows (`/Volumes/GoogleDrive/My Drive/ELISAarrayReader/data_for_tests_and_github/subset_for_tests`).

jennyfolkesson committed:
updated to pytest, tested workflows

jennyfolkesson committed:
added code coverage

jennyfolkesson committed:
merging

jennyfolkesson committed:
added codecov yml

jennyfolkesson committed:
removed xml from codecov

jennyfolkesson committed:
added code coverage badge

jennyfolkesson commented:
@mattersoflight I've rebased wrt master and checked that it works for array_fit and array_interp.
I checked that it ran for the well_* workflows but not sure what they're trying to do so I can't say they're working, just that they didn't crash. :-)
I also changed existing unittests to pytests.
I didn't pick images, figured that can be saved for another day/branch...
I also added code coverage, so we can see how testing will be getting better over time hopefully.

Please take a look again. Thanks!

mattersoflight commented on array_analyzer/utils/io_utils.py:
```diff
@@ -0,0 +1,97 @@
+    for ext in extensions:
+        search_str = os.path.join(input_dir, '**', ext)
+        image_names.extend(glob.glob(search_str, recursive=True))
+
```
searching the folder tree recursively returns too many images, and assigns wrong images to the wells.
<img width="1441" alt="image" src="https://user-images.githubusercontent.com/2934183/80677235-722f6b00-8a6d-11ea-9757-2bf7cf3acb13.png">

mattersoflight replied:
Above screenshot was taken with following arguments: 
`-e
-i="/Volumes/GoogleDrive/My Drive/ELISAarrayReader/images_scienion/2020-04-08-14-48-49-COVID_concentrationtest_April8_images"
-o="~/Documents/images_local/2020-04-08-14-48-49-COVID_concentrationtest_April8_images"
-wf=array_fit`

jennyfolkesson replied:
When I tested it on the data paths I didn't get these issues, but I never write outputs in the same directory as the input so that's probably why I missed that. Thanks for not only catching it but also fixing it!

bryantChhun replied:
> When I tested it on the data paths I didn't get these issues, but I never write outputs in the same directory as the input so that's probably why I missed that. Thanks for not only catching it but also fixing it!

Somewhat related -- do we want to generally allow writing reports and outputs to the same directory as the raw data files?  It does cause problems when, for example, searching for excel metadata spreadsheets.

mattersoflight reviewed (changes requested):
The discovery of image files is little buggy

mattersoflight committed:
get_image_paths: make file search more robust

mattersoflight commented:
I've fixed the bug by restricting the filename search to either the input folder or one level down from input folder. The well workflow runs fine now and array_fit workflow runs too, but I don't see an output OD file.

@jennyfolkesson  please take a look.

jennyfolkesson committed:
changed od name to match fit_interp

jennyfolkesson commented:
@mattersoflight I did get an OD file, but I named it similar to the debug excel files. I've now renamed the OD excel file to match the one in the interpolation workflow: 'python_median_ODs.xlsx'. I hope that helps.

mattersoflight committed:
make filenames of output consistent with interpolation workflow.

mattersoflight commented:
I changed the other two filenames to be consistent too. Merging.
